src/riverroute/RtmTimeManager.F90

```diff
@@ -1001,7 +1001,7 @@ logical function is_first_step()
    call ESMF_ClockGet( tm_clock, advanceCount=step_no, rc=rc )
    call chkrc(rc, sub//': error return from ESMF_ClockGet')
    nstep = step_no
-   is_first_step = (nstep == 0)
+   is_first_step = (nstep == 1)
```

Comment:
Don't we also need to change behavior where ever this function is used?  This doesn't seem to stop running the 0th timestep instead it's just ignoring it instead?

Reply:
@slevis-lmwg this is a good question and something that we should make sure is correct. I suggest we meet with @olyson and you and I to make sure this is right.

Reply:
Keith, I'm interested in your feedback. If a meeting is necessary, the two of us can meet first and include Erik if we need his support.

Reply:
As I go through the corresponding ctsm PR https://github.com/ESCOMP/CTSM/pull/2084
I see that you made the same change to function is_first_step in clm_time_manager.F90. So this discussion pertains to both.

Reply:
Another clue:
The function's docstring says "Return true on first step of initial run only" but now it will return true for all RUN_TYPEs. This makes me think that we do need to see where and why the function is called.

Reply:
...but also, the more I go through your code mods, the more I see that you consciously either removed or kept calls to this function depending on the case. So more likely we just need to reword the function's docstring.

Reply:
Since the function is_first_step has now been removed in mosart, it seems like it could just be removed here as well.  I don't see that the function is used anywhere.  We might have changed the line of code in that function in case the function was ever used.  At least that's my recollection, the PR was issued a year and a half ago....
Regarding where CLM stops running the 0th timestep, that is done in src/utils/clm_time_manager.F90 by calling advance_timestep.
We carefully went through all the calls to this function per the discussion in https://github.com/ESCOMP/CTSM/issues/925.
I'm not sure where the "docstring" is....
Maybe we can discuss further at our meeting tomorrow.

Reply:
Thanks, Keith. I have decided to just update the docstring of `is_first_step` in both rtm and ctsm (see my most recent commits). Although the function is not used in rtm, I will not remove it at this time.